python/src/trezorlib/ethereum.py

```diff
@@ -83,6 +89,49 @@ def sign_tx(
     return response.signature_v, response.signature_r, response.signature_s
 
 
+@session
+def sign_tx_eip1559(
+    client,
+    n,
+    nonce,
+    gas_limit,
+    to,
+    value,
+    data=None,
+    chain_id=None,
+    max_gas_fee=None,
+    max_priority_fee=None,
+    access_list=(),
+):
+    msg = messages.EthereumSignTxEIP1559(
+        address_n=n,
+        nonce=int_to_big_endian(nonce),
+        gas_limit=int_to_big_endian(gas_limit),
+        value=int_to_big_endian(value),
+        to=to,
+        chain_id=chain_id,
+        max_gas_fee=int_to_big_endian(max_gas_fee) if max_gas_fee is not None else None,
+        max_priority_fee=int_to_big_endian(max_priority_fee)
+        if max_priority_fee is not None
+        else None,
+        access_list=access_list,
+    )
+
+    if data:
+        msg.data_length = len(data)
+        data, chunk = data[1024:], data[:1024]
+        msg.data_initial_chunk = chunk
+
+    response = client.call(msg)
+
+    while response.data_length is not None:
+        data_length = response.data_length
+        data, chunk = data[data_length:], data[:data_length]
+        response = client.call(messages.EthereumTxAck(data_chunk=chunk))
+
+    return response.signature_v, response.signature_r, response.signature_s
```

Comment:
Given that you don't use `chain_id` in the signature, I think you will need to do the appropriate modification here. (per EIP-155 the sig_v must be `sig_byte + chain_id * 2 + 35`)

Or does EIP1559 invalidate that?

Reply:
`chain_id` is part of the EIP 1559 tx payload and is therefore not packed in `v` anymore. This is intended.

Reply:
i see! it's not even `sig_v` anymore. thanks.